wallet/storage/src/internal/mod.rs

```diff
@@ -31,19 +39,138 @@ use wallet_types::{
     AccountTxId, KeychainUsageState, RootKeyContent, RootKeyId, WalletTx,
 };
 
+use self::store_tx::EncryptionState;
+
+fn challenge_to_sym_key(
+    password: &String,
+    kdf_challenge: KdfChallenge,
+) -> Result<SymmetricKey, crate::Error> {
+    let KdfResult::Argon2id {
+        hashed_password, ..
+    } = hash_from_challenge(kdf_challenge, password.as_bytes())
+        .map_err(|_| crate::Error::WalletInvalidPassword())?;
+
+    let sym_key = SymmetricKey::from_raw_key(
+        SymmetricKeyKind::XChacha20Poly1305,
+        hashed_password.as_slice(),
+    )
+    .expect("must be correct size");
+
+    Ok(sym_key)
+}
+
+fn password_to_sym_key(password: &String) -> crate::Result<(SymmetricKey, KdfChallenge)> {
+    let mut rng = make_true_rng();
+    let config = KdfConfig::Argon2id {
+        // TODO: hardcoded values
+        config: Argon2Config::new(700, 16, 2),
+        hash_length: NonZeroUsize::new(32).expect("not 0"),
+        salt_length: NonZeroUsize::new(32).expect("not 0"),
+    };
+    let kdf_result = hash_password(&mut rng, config, password.as_bytes())
+        .map_err(|_| crate::Error::WalletInvalidPassword())?;
+    let KdfResult::Argon2id {
+        hashed_password, ..
```

Comment:
Please don't use `..` at all. Generally speaking, we want the compiler to yell at us when we change something, so that everything can be thought of when invariants are broken.

Reply:
fixed